src/Tizen.NUI/src/public/ParticleSystem/ParticleEmitter.cs

```diff
@@ -0,0 +1,577 @@
+            modifier.SetEmitter(this);
+            
+            Interop.ParticleEmitter.AddModifier(SwigCPtr, modifier.SwigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Returns associated ParticleSource object
+        /// </summary>
+        /// <returns>Valid ParticleSource object or null</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ParticleSource<T> GetSource<T>() where T : ParticleSourceInterface, new()
+        {
+            IntPtr cPtr = Interop.ParticleEmitter.GetSource(SwigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            ParticleSource<T> ret = (cPtr == IntPtr.Zero) ? null : Registry.GetManagedBaseHandleFromNativePtr(cPtr) as ParticleSource<T>;
+            return ret;
+        }
+
+        /// <summary>
+        /// Returns modifier at specified index
+        /// </summary>
+        /// <param name="index">Index within modifier stack</param>
+        /// <returns>Valid ParticleModifier object or null</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ParticleModifier<ParticleModifierInterface> GetModifierAt(uint index)
+        {
+            IntPtr cPtr = Interop.ParticleEmitter.GetModifierAt(SwigCPtr, index);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            ParticleModifier<ParticleModifierInterface> ret = (cPtr == IntPtr.Zero) ? null : Registry.GetManagedBaseHandleFromNativePtr(cPtr) as ParticleModifier<ParticleModifierInterface>;
+            return ret;
+        }
+
+        /// <summary>
+        /// Starts emission of particles.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Start()
+        {
+            Interop.ParticleEmitter.Start(SwigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Stops emission of particles.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Stop()
+        {
+            Interop.ParticleEmitter.Stop(SwigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+        
+        /// <summary>
+        /// Adds local (not used by shader) data stream to the particle emitter
+        /// </summary>
+        /// <remarks>
+        /// Adds new stream of float type.
+        /// </remarks>
+        /// <param name="defaultValue">Default value to fill the stream with</param>
+        /// <returns>Index of newly created data stream</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public unsafe uint AddLocalStreamFloat(float defaultValue)
```

Comment:
I dont want to make some unsafe API on C# APIs. Couldnt you make these APIs seperate as 4 API,(with `ref float defaultValue` or something like that), and do some unsafe things on dali-csharp-binder side?

Currently, we are call some unified unsafe API, and I think we can avoid this unsafe behavior.

Reply:
I think I don't get how to do this. How can I remove `unsafe` when I should pass a pointer to an Interop function? Did you imply that I should change csharp binder code to not get pointer argument?

Reply:
That's what I just want to implement ways. I saw your csharp binder patches. It will works well I think